doc/sphinx/source/developer_guide/abi.rst

```diff
@@ -0,0 +1,99 @@
+.. developer_guide abi
+
+=============
+ABI messaging
+=============
+
+Paparazzi use a custom publish/subscribe middleware to exchange data between software components named ``ABI``.
+
+.. note:: This middleware was named *ABI* for *AirBorne Ivy*
+
+General Idea
+------------
+
+* Give an easy way to allow software components to exchange data with minimum delay nor execution overhead.
+* The components don't need to know each other, they only need the message format.
+* Each subscriber set a callback function that will be called when new data are sent.
+
+
+Message definition
+------------------
+
+The messages are described in ``conf/abi.xml`` analoguous to other messages in paparazzi. Name are unique, IDs starts from 0. "type" field should be a real C type. Field name might be useless.
+
+.. code-block:: xml
+
+    <class name="airborne">
+     <message name="DATA" id="0">
+      <field name="a" type="float"/>
+      <field name="b" type="struct bla"/>
+     </message>
+     ...
+    </class>
+
+Airborne code for subscriber
+----------------------------
+
+Include header and declare an ``abi_event`` as a global variable. Write the callback function with the proper prototype, matching the message.
+
+.. code-block:: C
+
+    #include "subsystems/abi.h"
+
+    abi_event ev;
+
+    void data_cb(uint8_t sender_id, const float * a, const struct bla * b) {
+     // do something here
+    }
+
+In the initialization function (or later) call the binding function for the message you want to receive.
+
+.. code-block:: C
+
+    AbiBindMsgDATA(ABI_BROADCAST, &ev, data_cb);
+
+
+The first parameter is the sender ID you want to receive the message from.
+
+* ``ABI_BROADCAST`` is used to receive messages from all senders.
+* ``ABI_DISABLE`` disable the callback (it will never be called).
+* Senders IDs can be found in the file ``sw/airborne/subsystems/abi_sender_ids.h``
+
+The second parameter is a pointer to the global ``abi_event`` you declared. This variable **can't** be reused for another bind. You must declare one abi_event per bind.
+
+The last parameter is your callback.
+
```

Comment:
add a note to tell that the "module creator" can help to prepare bindings when creating a new module ?